packages/cli/src/util/telemetry/index.ts

```diff
@@ -227,7 +230,79 @@ export class TelemetryEventStore {
     }
 
     if (this.enabled) {
-      // send events to the server
+      const sessionId = this.events[0].sessionId;
+      if (!sessionId) {
+        output.debug('Unable to send metrics: no session ID');
+        return;
+      }
+      const events = this.events.map(event => {
+        delete event.sessionId;
+        const { eventTime, teamId, ...rest } = event;
+        return { event_time: eventTime, team_id: teamId, ...rest };
+      });
+      const payload = {
+        headers: {
+          'Client-id': 'vercel-cli',
+          'x-vercel-cli-topic-id': 'generic',
+          'x-vercel-cli-session-id': sessionId,
+        },
+        body: events,
+      };
+      await this.sendToSubprocess(payload, output.debugEnabled);
+    }
+  }
+
+  /**
+   * Send the telemetry events to a subprocess, this invokes the `telemetry flush` command
+   * and passes a stringified payload to the subprocess, there's a risk that if the event payload
+   * increases in size, it may exceed the maximum buffer size for the subprocess, in which case the
+   * child process will error and not send anything.
+   * FIXME: handle max buffer size
+   */
+  async sendToSubprocess(payload: object, outputDebugEnabled: boolean) {
+    const args = [process.execPath, process.argv[0], process.argv[1]];
+    if (args[0] === args[1]) {
+      args.shift();
+    }
+    const nodeBinaryPath = args[0];
+    const script = [
+      ...args.slice(1),
+      'telemetry',
+      'flush',
+      JSON.stringify(payload),
+    ];
+    // When debugging, we want to know about the response from the server, so we can't exit early
+    if (outputDebugEnabled) {
+      return new Promise<void>(resolve => {
+        const childProcess = spawn(nodeBinaryPath, script, {
+          stdio: ['ignore', 'pipe', 'pipe'],
+        });
+
+        childProcess.stderr.on('data', data => output.debug(data.toString()));
+        childProcess.stdout.on('data', data => output.debug(data.toString()));
+        childProcess.on('error', d => {
+          output.debug(d);
+        });
+
+        setTimeout(() => {
+          childProcess.kill();
+        }, 2000);
```

Comment:
What is this timeout here for? If it's intentional, let's add an explanation comment